chains/substrate/chain.go

```diff
@@ -34,3 +36,34 @@ type ProposalExecutor interface {
 func NewSubstrateChain(listener EventListener, writer ProposalExecutor, blockstore *store.BlockStore, config *SubstrateConfig, executor BatchProposalExecutor) *SubstrateChain {
 	return &SubstrateChain{listener: listener, writer: writer, blockstore: blockstore, config: config, executor: executor}
 }
+
+// PollEvents is the goroutine that polls blocks and searches Deposit events in them.
+// Events are then sent to eventsChan.
+func (c *SubstrateChain) PollEvents(ctx context.Context, sysErr chan<- error, msgChan chan []*message.Message) {
+	log.Info().Msg("Polling Blocks...")
+
+	startBlock, err := c.blockstore.GetStartBlock(
+		*c.config.GeneralChainConfig.Id,
+		c.config.StartBlock,
+		c.config.GeneralChainConfig.LatestBlock,
+		c.config.GeneralChainConfig.FreshStart,
+	)
+
+	if err != nil {
+		sysErr <- fmt.Errorf("error %w on getting last stored block", err)
+		return
+	}
+
```

Comment:
Please add this https://github.com/sygmaprotocol/sygma-relayer/blob/17a517f1f9ad1c4d27793e80bf762ca5fd103a64/chains/evm/chain.go#L86 here.